### PR TITLE
Bug 1604387 - Improve QueryScorer.

### PR DESCRIPTION
**Pull request by 0c0w3**
Please see bug 1604387. This patch fixes the first problem by
matching on whole words only for the first word in the query. For
subsequent words, it matches on prefixes. Prefix matching is done
the same way as whole word matching, by edit distance. So words
in the query can still contain typos while still matching. I
precompute all possible prefixes so that query matching is still
efficient (lookup in a map in constant time).

This fixes the second problem by making "firefox" and "mozilla"
stop words. i.e., the user can type them and they won't match
anything, but they also won't prevent matches.

I think this works pretty well.

Other changes:

* Bump the version number.
* resetProfile.xul and sanitize.xul were recently changed to
  xhtml, so I modified the test to account for that, keeping in
  mind that this experiment still targets 72.

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -41,16 +41,12 @@ const KEYWORDS = {
     "2019",
```
Comment (mak77) [inline src/background.js:41]: we are very close to 2020 and the experiment will run in 2020
we should maybe also add "www.getfirefox.com if we have "www.firefox.com"

**Review by mak77 — COMMENTED** [review]
Do I read correctly that this may trigger the update intervention on things like "free download", "latest download", "get mac", "new mac", "get free", "download windows"... Is that something expected? to me it would look like aggressive advertising.
The other interventions are less problematic, I could think of "slow working" as a case that doesn't match that well, maybe the match should be against "not work". "work" may be a very common word for people to type for a lot of reasons.

In general, I don't like the way we match, I think we should first ensure the phrase is mozilla or firefox related (check that among all the words there's either "moz", "firefox", "fox fire", "fire fox" with 1 typo admitted), and then remove the matched mozilla word and check the rest against a words list.
Could you please confirm with Verdi that it's fine to show an update intervention for some of the combos I suggested above in the experiment? I mean, maybe we also want to test negative feedback from users.

If I misunderstood the matching, please let me know.

**Comment by 0c0w3** [comment]
We already match on those queries even before this patch. I've asked Verdi.

**Comment by 0c0w3** [comment]
Abandoning this in favor of #8
